pkg/operator/controller/externaldns/deployment.go

```diff
@@ -598,3 +602,78 @@ func buildStringMapHash(data map[string]string) (string, error) {
 	}
 	return buildMapHash(m)
 }
+
+func securityContextChanged(current, updated, desired *corev1.SecurityContext) (bool, *corev1.SecurityContext) {
+	desiredCopy := desired.DeepCopy()
+	updatedFlag := false
+
+	if desiredCopy == nil {
+		return false, nil
+	}
+
+	if updated == nil {
+		return true, desiredCopy
+	}
+
+	if current == nil {
+		return true, updated
+	}
+
+	if desiredCopy.Capabilities != nil {
+		cmpCapabilities := cmpopts.SortSlices(func(a, b corev1.Capability) bool { return a < b })
+		if current.Capabilities == nil {
+			updated.Capabilities = desiredCopy.Capabilities
+			updatedFlag = true
+		} else {
+			if !cmp.Equal(desiredCopy.Capabilities.Add, current.Capabilities.Add, cmpCapabilities) {
+				copy(updated.Capabilities.Add, desiredCopy.Capabilities.Add)
```

Comment:
`copy` won't work in all the cases: if the `updated.Capabilities.Add` is smaller then desired one, not all desired capabilities will be copied. A simple assignment will be fine here.
```suggestion
				updated.Capabilities.Add = desiredCopy.Capabilities.Add
```